pallets/authors-manager/src/lib.rs

```diff
@@ -0,0 +1,610 @@
+    scale_info::TypeInfo,
+    traits::{Convert, Member},
+    DispatchError,
+};
+use sp_std::prelude::*;
+
+use pallet_avn::{
+    self as avn, AccountToBytesConverter, BridgeInterface, BridgeInterfaceNotification,
+    NewSessionHandler, ValidatorRegistrationNotifier,
+};
+use sp_avn_common::{
+    eth_key_actions::decompress_eth_public_key, event_types::Validator, IngressCounter,
+};
+
+pub use pallet::*;
+
+const PALLET_ID: &'static [u8; 14] = b"author_manager";
+
+const DEFAULT_MINIMUM_AUTHORS_COUNT: usize = 2;
+
+pub mod default_weights;
+pub use default_weights::WeightInfo;
+
+pub type AVN<T> = avn::Pallet<T>;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::{assert_ok, pallet_prelude::*};
+    use frame_system::{offchain::SendTransactionTypes, pallet_prelude::*};
+    use pallet_avn::{EthereumPublicKeyChecker, MAX_VALIDATOR_ACCOUNTS};
+    use sp_avn_common::bounds::MaximumValidatorsBound;
+    use sp_core::ecdsa;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config:
+        SendTransactionTypes<Call<Self>>
+        + frame_system::Config
+        + session::Config
+        + pallet_avn::Config
+        + pallet_session::historical::Config
+    {
+        /// Overarching event type
+        type RuntimeEvent: From<Event<Self>>
+            + Into<<Self as frame_system::Config>::RuntimeEvent>
+            + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        type AccountToBytesConvert: AccountToBytesConverter<Self::AccountId>;
+
+        type ValidatorRegistrationNotifier: ValidatorRegistrationNotifier<
+            <Self as session::Config>::ValidatorId,
+        >;
+
+        type WeightInfo: WeightInfo;
+
+        type BridgeInterface: BridgeInterface;
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        AuthorRegistered { author_id: T::AccountId, eth_key: ecdsa::Public },
+        AuthorDeregistered { author_id: T::AccountId },
+        AuthorActivationStarted { author_id: T::AccountId },
+        AuthorActionConfirmed { action_id: ActionId<T::AccountId> },
+        PublishingAuthorActionOnEthereumFailed { tx_id: u32 },
+        PublishingAuthorActionOnEthereumSucceeded { tx_id: u32 },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        NoTier1EventForAddingAuthor,
+        NoTier1EventForRemovingAuthor,
+        NoAuthors,
+        AuthorAlreadyExists,
+        InvalidIngressCounter,
+        MinimumAuthorsReached,
+        ErrorEndingVotingPeriod,
+        VotingSessionIsNotValid,
+        ErrorSubmitCandidateTxnToTier1,
+        ErrorCalculatingPrimaryAuthor,
+        AuthorsActionDataNotFound,
+        RemovalAlreadyRequested,
+        ErrorConvertingAccountIdToAuthorId,
+        SlashedAuthorIsNotFound,
+        AuthorNotFound,
+        InvalidPublicKey,
+        /// The ethereum public key of this author already exists
```

Comment:
why is this the only Error with a comment?

Reply:
It was copied from the original pallet. But I have updated it now.